src/shared-validator.h

```diff
@@ -169,10 +173,23 @@ class SharedValidator {
 
  private:
   struct FuncType {
+    FuncType() = default;
+    FuncType(const TypeVector& params, const TypeVector& results)
+        : params(params), results(results) {}
+
     TypeVector params;
     TypeVector results;
   };
 
+  using TypeMutVector = std::vector<TypeMut>;
```

Comment:
Nit: this seems dubiously-useful, and we still have two `std::vector<TypeMut>`s in this PR. I'd either move this to be next to `TypeMut`'s declaration so it's usable everywhere, or just use naked std::vectors.

Reply:
Good point, I moved it to where `TypeMut` is defined, and used it everywhere. I like it a bit better than using `std::vector` everywhere.